templates/common/ovirt/units/nodeip-configuration.service.yaml

```diff
@@ -0,0 +1,68 @@
+name: nodeip-configuration.service
+enabled: {{if .Infra -}}
+         {{if .Infra.Status -}}
+         {{if .Infra.Status.PlatformStatus -}}
+         {{if .Infra.Status.PlatformStatus.Ovirt -}}
+         {{if .Infra.Status.PlatformStatus.Ovirt.APIServerInternalIP -}}
+         true
+         {{- else -}}
+         false
+         {{- end -}}
+         {{- else -}}
+         false
+         {{- end -}}
+         {{- else -}}
+         false
+         {{- end -}}
+         {{- else -}}
+         false
+         {{- end -}}
+         {{- else -}}
+         false
+         {{- end }}
+contents: |
+  [Unit]
+  Description=Writes IP address configuration so that kubelet and crio services select a valid node IP
+  # This only applies to VIP managing environments where the kubelet and crio IP
+  # address picking logic is flawed and may end up selecting an address from a
+  # different subnet or a deprecated address
+  Wants=network-online.target
+  After=network-online.target ignition-firstboot-complete.service
+  Before=kubelet.service crio.service
+
+  [Service]
+  # Need oneshot to delay kubelet
+  Type=oneshot
+  # Would prefer to do Restart=on-failure instead of this bash retry loop, but
+  # the version of systemd we have right now doesn't support it. It should be
+  # available in systemd v244 and higher.
+  ExecStart=/bin/bash -c " \
+    until \
+    /usr/bin/podman run --rm \
+    --authfile /var/lib/kubelet/config.json \
+    --volume /etc/systemd/system:/etc/systemd/system:z \
+    --net=host \
+    {{ .Images.baremetalRuntimeCfgImage }} \
+    node-ip \
+    set --retry-on-failure \
+    {{.Infra.Status.PlatformStatus.Ovirt.APIServerInternalIP }}; \
```

Comment:
Does this attribute exist in UPI? We had the conditionals to protect against an exception in UPi where this value does not exist.

Reply:
Nope, it wouldn't exist, but I thought that if the service is disabled it doesn't matter.
That is not the case?

Reply:
The template is written regardless, so without the condition the mco will throw an exception trying to access the value while writing the template.

Reply:
thanks for the explanation, fixed